src/panels/developer-tools/state/developer-tools-state.ts

```diff
@@ -0,0 +1,682 @@
+import { addHours } from "date-fns/esm";
+import "@material/mwc-button";
+import {
+  mdiClipboardTextMultipleOutline,
+  mdiInformationOutline,
+  mdiRefresh,
+} from "@mdi/js";
+import { CSSResultGroup, LitElement, css, html, nothing } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import { formatDateTimeWithSeconds } from "../../../common/datetime/format_date_time";
+import { computeRTL } from "../../../common/util/compute_rtl";
+import { escapeRegExp } from "../../../common/string/escape_regexp";
+import { copyToClipboard } from "../../../common/util/copy-clipboard";
+import "../../../components/entity/ha-entity-picker";
+import "../../../components/ha-yaml-editor";
+import "../../../components/ha-icon-button";
+import "../../../components/ha-svg-icon";
+import "../../../components/ha-checkbox";
+import "../../../components/ha-tip";
+import "../../../components/ha-alert";
+import "../../../components/search-input";
+import "../../../components/ha-expansion-panel";
+import { showAlertDialog } from "../../../dialogs/generic/show-dialog-box";
+import { haStyle } from "../../../resources/styles";
+import { HomeAssistant } from "../../../types";
+import { fireEvent } from "../../../common/dom/fire_event";
+import { toggleAttribute } from "../../../common/dom/toggle_attribute";
+
+@customElement("developer-tools-state")
+class HaPanelDevState extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() private _error: string = "";
+
+  @state() private _entityId: string = "";
+
+  @state() private _entityFilter: string = "";
+
+  @state() private _stateFilter: string = "";
+
+  @state() private _attributeFilter: string = "";
+
+  @state() private _entity?: object;
+
+  @state() private _state: string = "";
+
+  @state() private _stateAttributes: object = {};
+
+  @state() private _expanded: boolean = false;
```

Comment:
Minor TS suggestion: remove the `: boolean`.  TS automatically infers the type from the assignment so it's redundant.